src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs

```diff
@@ -238,6 +238,7 @@ protected internal virtual void OnMessageAddedToPoisonQueue(PoisonMessageEventAr
             MessageAddedToPoisonQueue?.Invoke(this, e);
         }
 
+        // TODO the same as StorageQueueExtensions
```

Comment:
?

Reply:
https://github.com/Azure/azure-webjobs-sdk/blob/e648136df61a6f43ac7bc139739e24df9f2fbfef/src/Microsoft.Azure.WebJobs.Host/Queues/StorageQueueExtensions.cs#L16-L44

and it's twin brother

https://github.com/Azure/azure-webjobs-sdk/blob/e648136df61a6f43ac7bc139739e24df9f2fbfef/src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs#L241-L267

but it looks like the `poison queue` don't have an abstraction (take `CloudQueue` as argument)? my guess is that since `poison queue` is always associated with queuelistener, it only have end to end test